docs/source/api/utils/linalg.rst

```diff
@@ -0,0 +1,6 @@
+probnum.utils.linalg
```

Comment:
Souldn't these go under linalg API? I.e., `pobnum.linalg.utils` instead of `probnum.utils.linalg`? It does't seem right to move sth specific to a general package.

Reply:
Possibly yes. This should be a separate refactoring issue however. Since this is up in the air, I removed the `utils` part of the refactoring.

Reply:
Sure, we can discuss that later as well. It might be easier then to leave the utils part in this PR since it is easy to see which util function belong to `linalg`. and we can still see later if we want to move it. Does that make sense?

Reply:
The idea behind moving the Cholesky updates to (the thereby created) `utils.linalg` instead of `linalg.utils` was (a) Cholesky updates are a linear algebra tool for PN algorithms, and not a probabilistic linear solver, and (b) they are needed in e.g. `statespace`, and a linalg dependency for this small thing would've been a bit weird in my opinion. If you have stronger reasons to move it there, though, I am happy to discuss :)

Reply:
I see, then simply the namespace is confusing as is may imply that the `utils.linalg` are only required by `linalg` subpackage. If the utils are used by multiple subpackages they should stay where they are (perhaps with a different namespace.) Thanks for the explanation @pnkraemer !